### PR TITLE
deposit: datacite DOI minting

### PR DESCRIPTION
**Pull request by krzysztof**
* Adds signal for datacite DOI minting.

Signed-off-by: Krzysztof Nowak <k.nowak@cern.ch>

### CODE REVIEW

File: requirements.devel.txt [hunk 1]
```diff
@@ -23,7 +23,7 @@
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
 # Requirements for testing development version of important packages.
--e git+https://github.com/inveniosoftware/invenio-deposit.git#egg=invenio-deposit
+-e git+https://github.com/krzysztof/invenio-deposit.git@post_action_signal#egg=invenio-deposit
```
Comment (lnielsen) [inline requirements.devel.txt]: rm

File: zenodo/modules/deposit/tasks.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """Mint the DOI with DataCite.
+
+    :param pid_value: Value of record PID, with pid_type='recid'.
+    :type pid_value: str
+    """
+    record = Record.get_record(record_uuid)
+    dcp = DataCiteProvider.get(record['doi'])
+    url = current_app.config['ZENODO_RECORDS_UI_LINKS_FORMAT'].format(
+        recid=pid_value)
+    doc = datacite_v31.dump(dict(metadata=record.dumps()))
```
Comment (lnielsen) [inline zenodo/modules/deposit/tasks.py]: ``datacite_v31.serialize(pid, record)``

File: zenodo/modules/deposit/tasks.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    dcp.register(url, doc)
```
Comment (lnielsen) [inline zenodo/modules/deposit/tasks.py]:
```python
dcp.update(url, doc) if dip.pid.status == PIDStatus.REGISTERED else dcp.register(url, doc)
```

File: zenodo/modules/deposit/tasks.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+from celery import shared_task
+
+from invenio_pidstore.providers.datacite import DataCiteProvider
+from zenodo.modules.records.serializers import datacite_v31
+from invenio_records_files.api import Record
+from flask import current_app
+
+
+@shared_task
```
Comment (lnielsen) [inline zenodo/modules/deposit/tasks.py]: Retry: https://github.com/inveniosoftware/invenio/blob/maint-2.0/invenio/modules/pidstore/tasks/datacite.py#L169-L170

File: tests/unit/deposit/test_api_simpleflow.py [hunk 5]
```diff
@@ -48,9 +50,24 @@ def make_file_fixture(filename, text=None):
     return (BytesIO(content), filename)
 
 
-def test_simple_rest_flow(api, api_client, db, es, location, users,
+class MockDataCiteProvider(DataCiteProvider):
```
Comment (lnielsen) [inline tests/unit/deposit/test_api_simpleflow.py]: I think its better to mock ``invenio_pidstore.providers.datacite.DataCiteMDSClient`` instead. This way we insure that the DOI is marked as registered in PIDStore.

File: requirements.devel.txt (hunk 1 above)
Comment (lnielsen) [inline requirements.devel.txt]: I've merged the PR in invenio-deposit now, so this one can be removed.

File: zenodo/modules/deposit/tasks.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    :type pid_value: str
+    """
+    record = Record.get_record(record_uuid)
+    dcp = DataCiteProvider.get(record['doi'])
+    url = current_app.config['ZENODO_RECORDS_UI_LINKS_FORMAT'].format(
+        recid=pid_value)
+    pid = dcp.pid
+    doc = datacite_v31.serialize(pid, record)
+    dcp.update(url, doc) if dcp.pid.status == PIDStatus.REGISTERED \
+        else dcp.register(url, doc)
```
Comment (lnielsen) [inline zenodo/modules/deposit/tasks.py]: ``db.session.commit()``?

Comment (krzysztof) [inline zenodo/modules/deposit/tasks.py]: This will probably close the transaction from the task on CELERY_ALWAYS_EAGER.

**Comment by krzysztof** [comment]
This pins Flask-Admin to 1.4.0 (related issue at https://github.com/zenodo/zenodo/issues/604)

**Pull request merged**